File: src/basic_training/Day09_4.java
package basic_training;

import java.util.Arrays;

public class Day09_4 {
	public String[] solution(String my_String) {
		
		String[] answer = new String[my_String.length()];
		for(int i=0; i<my_String.length(); i++){
			answer[i] = my_String.substring(i);
		}
		Arrays.sort(answer);
		return answer;
	}
}

// 자바 배열 정렬 _ Object 타입 (String, Integer, Double)
// Arrays.sort(배열);

// 내림차순 정렬
// Arrays.sort(배열, Collections.reverseOrder());

// 기본타입인 int, double, char, float 등은 Wrapper 클래스로 박싱해주어야 정렬 가능
// 예) int[] -> Integer[]

File: src/basic_training/Day09_4.js
function solution(my_string) {
	
	return my_string.split('').map((ele, idx)=> my_string.substring(idx)).sort();
}

// my_string.split(''); // ['b','a','n','a','n','a']
// idx == 0 => my_string.substring(0) == 'banana'
//(왜냐? my_string 은 'banana' 이기 때문에 😋)

// 자바스크립트 배열.slice(시작인덱스, 마지막인덱스(포함x)); 
// 자바스크립트 배열.substring(시작인덱스, 마지막인덱스(포함x)); 
// 새로운 배열 반환

// 🌼 번외 문제풀이
// let answer = [];
// let arr = my_string.split(''); // ['b','a','n','a','n','a']

// for(let i=0; i<my_string.length; i++) {
// 	answer.push(arr.slice(i).join('')); // 'banana'
// }
// return answer.sort();